test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -349,6 +355,34 @@ protected void afterIfFailed(List<Throwable> errors) {}
     /** called after a test is finished, but only if successful */
     protected void afterIfSuccessful() throws Exception {}
 
+    /**
+     * Marks a test suite or a test method that should run without security manager enabled.
+     */
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target({ ElementType.TYPE })
+    @Inherited
+    public @interface NoSecurityManager {
```

Comment:
I wonder if we should place this on `ESClientYamlSuiteTestCase` since there's not point in _any_ yaml rest tests using the security manager. You could probably make an argument for java rest tests as well but those are a bit more of the wild west.

Reply:
Are ESClientYamlSuiteTestCases ever runnable in the IDE? I think it's likely better to just decouple those from ESTestCase, since as you say, they don't ever need to run with security manager.

Reply:
invoked via the gradle runner, those tests should run from the IDE

Reply:
I will look at what classes can utilize this annotation more in followups, I'd like to keep this PR as small as possible just for the infrastructure.